### PR TITLE
feat: WebApp allows model training; completes #37

### PR DESCRIPTION
**Pull request by ArttuLe**
- WebApp is able to train a human detection model from scratch and converts it into a TFLite model, with or without quantization, for the model binary generation in #38 .
- Temporary solution, but dataset is now stored in the repo.
- UI needs more work and code could be refactored a bit but it is working now.
- Model architecture might need to be changed, might be a bit heavy for TinyML usecases at the moment.
- See Discord for a gif.

### CODE REVIEW

**Comment by ArttuLe** [comment]
Still requires nbdev documentation code and tests(?) @doyu

**Comment by doyu** [comment]
> Still requires nbdev documentation code and tests(?) @doyu

Yes, please. Probably you need Acceptance test merged before this.

One note is that,
For training data, can we just assume that data is stored in the disk in cerntain format (e.g. parent dirname indicates label "0"/"1"). With this, we could get rid of TCP server out of TinyMLaaS but just read data from storage. How to accept and store data could be completely independent from TinyMLaaS, which could be done in another process.

Ever for observation of inference data (i.e. no label but just prediction), the captured image could be just stored in the parent dir, "2":"unknown", since we don't consider perf yet this point. If we need some perf, we could consdier other way, but for now, passing data via storage would be enough for our use case? This is related to other items, I suppose.

**Comment by ArttuLe** [comment]
> > Still requires nbdev documentation code and tests(?) @doyu
> 
> Yes, please. Probably you need to implement "Acceptance test" in RF and it should be merged before this change, in the context of ATDD.
> 
> One note is that, For training data, can we just assume that data is stored in the disk in cerntain format (e.g. parent dirname indicates label "0"/"1"). With this, we could get rid of TCP server out of TinyMLaaS but just read data from storage. How to accept and store data could be completely independent from TinyMLaaS, which could be done in another process.
> 
> Ever for observation of inference data (i.e. no label but just prediction), the captured image could be just stored in the parent dir, "2":"unknown", since we don't consider perf yet this point. If we need some perf, we could consdier other way, but for now, passing data via storage would be enough for our use case? This is related to other items, I suppose.

- Yes, currently the dataloader gets the labels from the parent directory's name, i.e. "1" has humans in the picture and "0" doesn't, so it doesn't need any labeling apart from storing the images in the right directories. Later the data storage could be moved to a database and it should still work the same.  Need for TCP server or some other implementation comes when/if we want to gather more data?

**Comment by doyu** [comment]
As long as you use a file to pass between processes, it should work with prediction (i.e. no label) in the same way. I was just concerned about over engineering here.

**Comment by ArttuLe** [comment]
Could this be merged without the RF tests done and come back to the acceptance tests later, since the code/data in this PR are possibly needed for next tasks by others?

**Comment by doyu** [comment]
Can you check the CI error?

RF could be deferredt, though

**Comment by ArttuLe** [comment]
It is a nbdev generated file. Should those even be in the repo if they are generated by nbdev commands?

**Comment by doyu** [comment]
@lifeofborna could you check the above CI error?

**Comment by lifeofborna** [comment]
> @lifeofborna could you check the following CI error? How should we fix this from nbdev perspective?

You should try to run nbdev_clean & nbdev_export then it should pass the CI. The error says: "!!!Remember to run nbdev_install_hooks"

**Review by doyu — APPROVED** [review]

**Comment by doyu** [comment]
image

Setting.ini ?
